app/components/container/form-command/component.js

```diff
@@ -69,6 +75,15 @@ export default Component.extend({
     set(this, 'instance.stdin', stdin);
   }),
 
+  scaleModeDidChange: observer('scaleMode', function() {
+    const scaleMode = get(this, 'scaleMode');
+    if ( scaleMode === 'job' || scaleMode === 'cronJob' ) {
+      set(this, 'service.restartPolicy', 'Never');
+    } else {
+      set(this, 'service.restartPolicy', 'Always');
```

Comment:
This will be triggered when users switch the scale mode. Like from `Deployment` to `Job`, it will set the `restartPolicy` to `Never` as `Always` is not available for Job. Users still can select `OnFailure` in command section.

If users switch `Job` to `Deployment`, it will re-set the `restartPolicy` to `Always` which is the default value. But users still can select `Never` or `OnFailure` for deployment in command section.

Reply:
Yeah that's the part that's weird, all the settings from the workload are generally preserved when you change types. But if you've explicitly click On Failure and then switch to Deployment <-> Job you get Never (or Always), even though On Failure is valid for both types.

So I would expect it to only change to the default if the current value is invalid for the new service type.